### ISSUE TITLE
DMA complete callback functionality dependent on simulation interface being used

### ISSUE THREAD

**Issue by rjames91** [SpiNNakerManchester/SpiNNFrontEndCommon#262]
Noticed that the record and notify function doesn't work when using recording with an application that doesn't use simulation.c

This is due to the DMA callbacks being registered in the simulation initialisation function. When this isn't called (but recording initialisation is) the following function is called during recording initialisation which just adds the recording complete callbacks to a list (the spin1 callback on DMA isn't setup).

`    simulation_dma_transfer_done_callback_on(
        RECORDING_DMA_COMPLETE_TAG_ID, _recording_dma_finished);`

So when the DMA operation is complete the relevant recording callback is never called and the circular buffers aren't incremented.

It would be safer to detect or block this from happening. I would like this no simulation interface + recording functionality as I'm trying to speed things up by using a DMA complete callback as I'm finding the memcpy function is generating large performance overhead.

**Comment by alan-stokes** [comment]
this is kinda implying that we need to revisit the abstractions down there. or have separate ones for time based and none time based support........

its worth noting that you could initlise simulation and not use the auto pause and resume functionality.........

**Comment by rowleya** [comment]
Yes, it shouldn't depend directly on simulation - I guess we should have a "simulation_initialised" check at which point if this returns false, it could then do a normal DMA callback.  Note that you need to handle the DMA callback in any case...

**Comment by alan-stokes** [comment]
having had robert come over, hes planning to use the sim interface calls, aka registering his callbacks as well, so this might be a none issue now. given you dont need to use all of sim.c jsut the bits you want.
